advanced_functionality/huggingface_byo_scripts_and_data/source/train.py

```diff
@@ -0,0 +1,192 @@
+
+
+def train(args):
+    logger.info("Loading tokenizer...\n")
+    global tokenizer
+    global model_name
+    model_name = args.model_name
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    logger.info("Loading pretrained model\n")
+    model = AutoModelForSeq2SeqLM.from_pretrained(model_name)
+    logger.info("Pretrained model loaded\n")
+
+    logger.info("Fetching and tokenizing data for training")
+    train_dataset = load_and_tokenize_dataset(
+        args.train_data_dir,
+        "train",
+        args.text_column,
+        args.target_column,
+        args.max_source,
+        args.max_target,
+    )
+
+    logger.info("Tokenizing data for training loaded")
+
+    eval_dataset = load_and_tokenize_dataset(
+        args.train_data_dir,
+        "validation",
+        args.text_column,
+        args.target_column,
+        args.max_source,
+        args.max_target,
+    )
+    test_dataset = load_and_tokenize_dataset(
+        args.train_data_dir,
+        "test",
+        args.text_column,
+        args.target_column,
+        args.max_source,
+        args.max_target,
+    )
+
+    logger.info("Defining training arguments\n")
+    training_args = Seq2SeqTrainingArguments(
+        output_dir=args.model_dir,
+        num_train_epochs=args.epoch,
+        per_device_train_batch_size=args.train_batch_size,
+        per_device_eval_batch_size=args.eval_batch_size,
+        learning_rate=args.lr,
+        warmup_steps=args.warmup_steps,
+        weight_decay=args.weight_decay,
+        logging_dir=args.log_dir,
+        logging_strategy=args.logging_strategy,
+        load_best_model_at_end=True,
+        adafactor=True,
+        do_train=True,
+        do_eval=True,
+        do_predict=True,
+        save_total_limit=3,
+        evaluation_strategy="epoch",
+        save_strategy="epoch",
+        predict_with_generate=True,
+        metric_for_best_model="eval_loss",
+        seed=7,
+    )
+
+    logger.info("Defining seq2seq Trainer")
+    trainer = Seq2SeqTrainer(
+        model=model,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        tokenizer=tokenizer,
+        compute_metrics=compute_metrics,
+    )
+
+    logger.info("Starting Training")
+    trainer.train()
+    logger.info("Model trained successfully")
+    trainer.save_model()
+    logger.info("Model saved successfully")
+
+    # Evaluation
+    logger.info("*** Evaluate on test set***")
+
+    logger.info(trainer.predict(test_dataset))
+
+    logger.info("Removing unused checkpoints to save space in container")
+    os.system(f"rm -rf {args.model_dir}/checkpoint-*/")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model-name", type=str, default="google/pegasus-xsum")
+    parser.add_argument(
+        "--train-data-dir", type=str, default=os.environ["SM_CHANNEL_TRAIN"]
+    )
+    # parser.add_argument("--val-data-dir", type=str,
```

Comment:
Can these `--val-data-dir` and `--test-data-dir` args be removed, or would a customer possibly want to uncomment and use them?

Reply:
A customer would possibly want to use them, so this is why we keep them here